### DISCUSSION TITLE
How to receive messages by parts?

### DISCUSSION THREAD

**Discussion by ZedZipDev** [rabbitmq/rabbitmq-dotnet-client#1117] [Q&A]
How to receive messages from queue by parts? 
For example, I have a Queue (N messages) and clients need to receive by n<N messages from the queue but do not receive messages twice. How to implement this approach? I.e. every client reads like window by n msgs, then next n msgs and so on.

**Comment by lukebakken** [comment]
https://www.rabbitmq.com/consumer-prefetch.html

An application could receive a message twice if something happens during the processing of a message. For instance, if you have only one consumer with a prefetch of 10, and while processing those 10 messages your application crashes before acknowledging the messages, they will be re-delivered when the application starts again.

If you will be processing messages in batches you should also look at the `multiple` field for acks - https://www.rabbitmq.com/confirms.html#consumer-acks-multiple-parameter

**Comment by ZedZipDev** [comment]
Yes, excluding some crashes etc.

I have n consumer applications which read the same queue (now it is one queue). That is my question:
**each of n clients should receive messages only once from the queue.**
